src/components/progressBar/index.tsx

```diff
@@ -0,0 +1,189 @@
+
+/**
+ * @description: Progress bar
+ * @example:https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/ProgressBarScreen.js
+ */
+interface Props {
+  /**
+   * The progress of the bar from 0 to 100
+   */
+  progress?: number;
+  /**
+   *  FullWidth Ui preset
+   */
+  fullWidth?: boolean;
+  /**
+   * Override container style
+   */
+  style?: StyleProp<ViewStyle>;
+  /**
+   * Progress color
+   */
+  progressColor?: string,
+  /**
+   * Custom element to render on top of the animated progress
+   */
+  customElement?: JSX.Element
+}
+export type ProgressBarProps = Props;
+
+interface State {
+  containerWidth?: number;
+}
+
+class ProgressBar extends PureComponent<Props, State> {
+  static displayName = 'ProgressBar';
+
+  static defaultProps: Partial<Props> = {
+    progress: 0
+  }
+
+  progressAnimation: Animated.Value;
+
+  constructor(props: Props) {
+    super(props);
+    this.progressAnimation = new Animated.Value(props.progress || 0);
+
+    this.state = {
+      containerWidth: undefined
+    };
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const {progress} = this.props;
+
+    if (prevProps.progress !== progress) {
+      this.animateProgress(progress);
+    }
+  }
+
+  getContainerWidth = (event: LayoutChangeEvent) => {
+    if (!this.state.containerWidth) {
+      this.setState({containerWidth: event.nativeEvent.layout.width});
+    }
+  };
+
+  animateProgress(toValue?: number) {
+    if (toValue) {
+      Animated.timing(this.progressAnimation, {
+        duration: 220,
+        easing: Easing.ease,
+        toValue,
+        useNativeDriver: true
+      }).start();
+    }
+  }
+
+  getAccessibilityProps() {
+    const {progress} = this.props;
+    if (progress) {
+      return {
+        accessible: true,
+        accessibilityLabel: `progress bar. ${Math.round(progress)}%`,
+        ...extractAccessibilityProps()
+      };
+    }
+  }
+
+  getContainerStyle() {
+    const {fullWidth} = this.props;
+    const containerHeight = fullWidth ? FULL_WIDTH_CONTAINER_HEIGHT : CONTAINER_HEIGHT;
+    const tabletContainerHeight = fullWidth ? TABLET_FULL_WIDTH_CONTAINER_HEIGHT : TABLET_CONTAINER_HEIGHT;
+    const inlineStyle = fullWidth ? null : styles.inlineContainer;
+
+    return {
+      ...inlineStyle,
+      height: Constants.isTablet ? tabletContainerHeight : containerHeight
+    };
+  }
+
+  getProgressStyle() {
+    const {fullWidth, progressColor} = this.props;
+    const borderRadius = fullWidth ? styles.fullWidthProgressBorderRadius : styles.inlineBorderRadius;
+    const progressStyle = {
+      right: this.state.containerWidth,
+      backgroundColor: progressColor || DEFAULT_COLOR
+    };
+
+    return {
+      ...borderRadius,
+      ...progressStyle
+    };
+  }
+
+  renderCustomElement() {
+    const {customElement} = this.props;
+
+    if (customElement) {
+      return React.cloneElement(customElement, {
+        style: [customElement.props.style, styles.progress]
```

Comment:
I don't customElement has `props.style` 
Are you sure you're doing this correctly?

Reply:
What do you mean? Use optional chaining?

Reply:
Not exactly.. 
Does `customElement` has `props` on it.. after all we're passing a JSX, no?

Reply:
Well, yes. JSX extends ReactElement, which has 'props'

Reply:
What is the use case you're thinking about?